http_service/http_handler.go

```diff
@@ -0,0 +1,122 @@
+package http_service
+
+import (
+	"context"
+	"github.com/daniyakubov/book_service_n/book_service"
+	"github.com/daniyakubov/book_service_n/consts"
+	"github.com/daniyakubov/book_service_n/models"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type HttpHandler struct {
+	bookService book_Service.BookService
+}
+
+func NewHttpHandler(bookService book_Service.BookService) HttpHandler {
+	return HttpHandler{
+		bookService: bookService,
+	}
+}
+
+func (h *HttpHandler) AddBook(c *gin.Context) {
+	ctx := context.Background()
+	var book models.Book
+	err := c.Bind(&book)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	id, err := h.bookService.AddBook(&ctx, &book, c.Query(consts.UserName), c.FullPath())
```

Comment:
```suggestion
	id, err := h.bookService.AddBook(context.Background(), &book, c.Query(consts.UserName), c.FullPath())
```

and you can remove the first line decleration